ddtrace/tracer/span.go

```diff
@@ -80,12 +80,14 @@ type span struct {
 	ParentID   uint64             `msg:"parent_id"`             // identifier of the span's direct parent
 	Error      int32              `msg:"error"`                 // error status of the span; 0 means no errors
 	SpanLinks  []ddtrace.SpanLink `msg:"span_links,omitempty"`  // links to other spans
+	SpanEvents []spanEvent        `msg:"span_events,omitempty"` // events produced related to his span
```

Comment:
We keep adding new fields to the span but I'm not sure if this will bite us in the future.

I don't know enough about the library yet, to have some solid suggestions. I just wanted to bring it up to be more conscience about it.

Reply:
What's your concern with new fields, @kakkoyun? (or with this one specifically?)

Reply:
In general, for an instrumentation library, we put too much stuff. Considering users use it in their hot-paths being conscience about the resource usage is a sane thing to do. Nobody would want enabling observability and all of a sudden everything gets significantly slower or heavier on the memory.